notebook/gateway/managers.py

```diff
@@ -543,6 +549,9 @@ def get_all_specs(self):
     def list_kernel_specs(self):
         """Get a list of kernel specs."""
         kernel_spec_url = self._get_kernelspecs_endpoint_url()
+
+        kernel_spec_url = self._get_endpoint_for_user_filter(kernel_spec_url)
```

Comment:
Rather than redo the url calculation, I'm inclined to bake this into `self.base_endpoint` directly within the constructor.  Since env values are static there's really no reason to not run with the query parameter baked in.